### DISCUSSION TITLE
Sources access level roles and security

### DISCUSSION THREAD

**Discussion by toomuchdesign** [Budibase/budibase#9544] [Help] [answered]
As far as I understand BudiBase provides 2 configurable access levels to each data source:

- **Read**: all users with this right can **read** **all fields** of **all table entries**
- **Write**: all users with this right can **create/update/delete** **all fields** of **all table entries**

Let's apply this concept to any example shown in [BusiBase University](https://www.youtube.com/watch?v=kQUYgrkwxb0&list=PLEel-MMkFaJltK_jpKouKTCRht49FflKO), like the doctor appointment list. Does it mean that a patient with write access to the appointment source is able to create/update/delete any existing appointment created by other users?

Even if the UI gets configured to NOT allow this, I assume it would be trivial for a malicious user to programmatically send out a REST call to BudiBase backend to do so. Or is there some extra security level I've overlooked so far?

Thanks SO MUCH for your time, here!

**Comment by stearm** [comment]
any news about it?

**Comment by melohagan** [comment, marked as answer]
Hey guys,

This is correct. There is more RBAC work to be done to address these issues in future.

What you can do at the moment is to write to a table using [Automations](https://docs.budibase.com/docs/automation-steps)
That way the write access is protected from external API calls, and you would control which users can trigger the automation via the [screen access level](https://docs.budibase.com/docs/screens#access-level).

You would also set the role required for the automation to be triggered:
![Screenshot 2023-02-14 at 13 56 51](https://user-images.githubusercontent.com/101575380/218758958-dc9bedee-f2e5-4b1f-98c1-92371b6b2f9e.png)
